### ISSUE TITLE
StatusCode: 499

### ISSUE THREAD

**Issue by MatejKovacic** [bbernhard/signal-cli-rest-api#297]
### The problem

I have the latest version of signal-cli-rest-api. I run the command:
`curl -X GET -H "Content-Type: application/json" 'https://myserver.org/v1/receive/+XXXXXXXXXXX'` and get this error:

`{"error":"INFO  AccountHelper - The Signal protocol expects that incoming messages are regularly received.\nError while checking account +XXXXXXXXXXX: StatusCode: 499\n`

Also, I can not send messages.

### Are you using the latest released version?

- [X] Yes

### Have you read the troubleshooting page?

- [X] Yes

### What type of installation are you running?

signal-cli-rest-api Docker Container

### In which mode are you using the docker container?

Normal Mode

### What's the architecture of your host system?

x86-64

**Comment by gtmadev** [comment]
I just started having the same issue. Running dockerized. Been working for months and just now getting BAD REQUEST on all sends. From the logs, looks like this started on Oct 5

```
[GIN] 2022/10/03 - 23:00:01 | 200 |  1.388126743s |       127.0.0.1 | GET      "/v1/receive/+1307429XXXX"
[GIN] 2022/10/04 - 23:00:01 | 200 |  1.405387432s |       127.0.0.1 | GET      "/v1/receive/+1307429XXXX"
[GIN] 2022/10/05 - 23:00:00 | 400 |    308.8185ms |       127.0.0.1 | GET      "/v1/receive/+1307429XXXX"
time="2022-10-05T23:00:00Z" level=error msg="AUTO_RECEIVE_SCHEDULE: Couldn't call receive for number +1307429XXXX: {Error while checking account +1307429XXXX: StatusCode: 499\n}"
[GIN] 2022/10/06 - 23:00:00 | 400 |  538.226318ms |       127.0.0.1 | GET      "/v1/receive/+1307429XXXX"
time="2022-10-06T23:00:00Z" level=error msg="AUTO_RECEIVE_SCHEDULE: Couldn't call receive for number +1307429XXXX: {Error while checking account +1307429XXXX: StatusCode: 499\n}"
[GIN] 2022/10/07 - 23:00:00 | 400 |  303.266988ms |       127.0.0.1 | GET      "/v1/receive/+1307429XXXX"
time="2022-10-07T23:00:00Z" level=error msg="AUTO_RECEIVE_SCHEDULE: Couldn't call receive for number +1307429XXXX: {Error while checking account +1307429XXXX: StatusCode: 499\n}"

```

**Comment by MatejKovacic** [comment]
So i did:
`docker stop signal-api`

`docker pull bbernhard/signal-cli-rest-api`

I have the latest vesrion:
```
Using default tag: latest
latest: Pulling from bbernhard/signal-cli-rest-api
Digest: sha256:cbef6f022b090514dbd5072ab1a5dae9f011ac1d4fd22012e5acf3922b213e5b
Status: Image is up to date for bbernhard/signal-cli-rest-api:latest
```

`docker start signal-api`

`docker ps`

```
CONTAINER ID   IMAGE                               COMMAND                  CREATED        STATUS                            PORTS 
00c7cfd9e055   1c4b8bfaa013                        "/entrypoint.sh"         3 months ago   Up About a minute (healthy)   0.0.0.0:8111->8080/tcp, :::8111->8080/tcp                                                         signal-api
```
And when I try to send or receive a message:
```
{"error":"INFO  AccountHelper - The Signal protocol expects that incoming messages are regularly received.\nError while checking account +XXXXXXXXXXX: StatusCode: 499\n"}
```

**Comment by gtmadev** [comment]
I also saw that message in my logs prior to updating the image. I am not seeing it now, however I have the scheduler running only once daily. And it's not really something I am using. I just have it running at that interval because the docs recommend calling the receive command "on a regular basis". But my application does not rely on incoming messages, so they are ignored.

I will have to check the logs once the receive is called again.

**Comment by MatejKovacic** [comment]
OK, I run `docker exec -it signal-api /bin/bash` and there is no `/var/log/signal-cli-json-rpc-1/` folder. Also, `/var/log/supervisor/` directory is empty.

And, when I run `docker exec -it signal-api /bin/bash` I get these warnings:
```
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /home/.local/share/signal-cli"
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /home/.local/share/signal-cli"
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /home/.local/share/signal-cli"
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /home/.local/share/signal-cli"
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /home/.local/share/signal-cli"
root@00c7cfd9e055:/#
```

**Comment by bbernhard** [comment]
Please give this docker image tag a try: `bbernhard/signal-cli-rest-api:0.114-dev`

**Comment by MatejKovacic** [comment]
I did, but it is still the same.

`docker stop signal-api`

```
docker pull bbernhard/signal-cli-rest-api:0.114-dev

0.114-dev: Pulling from bbernhard/signal-cli-rest-api
...
Digest: sha256:814320f8932598d226054580270abf8adc58bda065a4cea56399ab16d23abbb7
Status: Downloaded newer image for bbernhard/signal-cli-rest-api:0.114-dev
docker.io/bbernhard/signal-cli-rest-api:0.114-dev
```

`docker start signal-api`

```
docker ps | grep signal-api

00c7cfd9e055   1c4b8bfaa013                        "/entrypoint.sh"         3 months ago   Up 2 minutes (healthy)   0.0.0.0:8111->8080/tcp, :::8111->8080/tcp                                                         signal-api
```

`curl` to the service still returns `StatusCode: 499`.

**Comment by bbernhard** [comment]
Can you please post the response from the `/about` endpoint.

**Comment by MatejKovacic** [comment]
`{"error":"INFO  AccountHelper - The Signal protocol expects that incoming messages are regularly received.\nError while checking account +XXXXXX: StatusCode: 499\n"}`

**Comment by MatejKovacic** [comment]
And I am trying to call `https://myserver.si/v1/receive/+XXXXXXXXX`

**Comment by hilbix** [comment]
Note that I had the same issue here.  I found out that my docker container did not update to the latest image for some reason.

But because of this I think the problem of @MatejKovacic might be the same.

To update the existing docker container, **`docker pull` alone is not enough**! 
To update an existing container the procedure is:

- docker stop
- docker pull
- docker rm
- docker run

Be sure to give all the needed options to `docker run` again.

Here is my startscript:

```
read API <~/.signal-clie-rest-api.version
sudo docker pull bbernhard/signal-cli-rest-api"$API"
sudo docker rm signal-api
exec sudo docker run -i --name signal-api --restart=always -p 127.0.0.1:8080:8080 -v $HOME/.local/share/signal-cli:/home/.local/share/signal-cli -e 'MODE=native' bbernhard/signal-cli-rest-api"$API"
```

FYI:

```
$ sudo docker images
REPOSITORY                      TAG         IMAGE ID       CREATED        SIZE
bbernhard/signal-cli-rest-api   0.114-dev   f4dbf8e04a4c   12 hours ago   660MB
bbernhard/signal-cli-rest-api   latest      cd6b090351f9   2 weeks ago    655MB

$ sudo docker container list
CONTAINER ID   IMAGE                                     COMMAND            CREATED         STATUS                   PORTS                      NAMES
36fa6b7304da   bbernhard/signal-cli-rest-api:0.114-dev   "/entrypoint.sh"   5 minutes ago   Up 5 minutes (healthy)   127.0.0.1:8080->8080/tcp   signal-api
```

PS: Thank you very much for your container.
PPS: Signal uses a landline number at my side, not a mobile phone one.  Works without problems.

**Comment by gtmadev** [comment]
Yeah, check the container's image hash. Do the following...

Get the container id.

Then run:   docker container inspect XXXXXXXXXX

Current Image (as of today) should be sha256:004e63a39b10b97d1971c24cae1b53f8e047c38d64cd36849cdb31c2bb0f07e4

**Comment by MatejKovacic** [comment]
Well, that was the problem. **I did not remove old container!** 

The old sha256 was `1c4b8bfaa013ff32d3f5c83570dbb9aa3779de67cff571fce5b91aa25d1327fb`, but after `docker rm signal-api` (and new start with `docker run -d --name signal-api ...`, everything is working fine.

Thank you for your help. Maybe just a suggestion - can you write a special section in `README.md` on how to update `signal-cli-rest-api` to a new version?

**Comment by bbernhard** [comment]
Cleaning up old issues. In case you still have questions/problems, please create another issue.
